core/coreapi/interface/interface.go

```diff
@@ -29,13 +30,13 @@ type Reader interface {
 type CoreAPI interface {
 	Unixfs() UnixfsAPI
 	ResolvePath(context.Context, Path) (Path, error)
-	ResolveNode(context.Context, Path) (Node, error)
+	ResolveNode(context.Context, Path) (Path, Node, error)
 }
 
 type UnixfsAPI interface {
 	Add(context.Context, io.Reader) (Path, error)
-	Cat(context.Context, Path) (Reader, error)
-	Ls(context.Context, Path) ([]*Link, error)
+	Cat(context.Context, Path) (Path, Reader, error)
+	Ls(context.Context, Path) (Path, []*Link, error)
```

Comment:
As much as I'm not a fan of mutable state, I'd say that we may want to not return resolved paths and just modify the ones passed to the methods - while it makes sense when we implement the API on top of go-ipfs node, doing this in http-api (or in any other wrapper that might be created) imo doesn't, and it would likely require changing the http api to return those resolved paths.